service/src/main/java/bio/terra/pipelines/app/controller/PipelineRunsApiController.java

```diff
@@ -164,6 +165,54 @@ public ResponseEntity<ApiAsyncPipelineRunResponse> getPipelineRunResult(
     return new ResponseEntity<>(runResponse, getAsyncResponseCode(runResponse.getJobReport()));
   }
 
+  /**
+   * Returns a paginated list of Pipeline Runs for the user calling this endpoint. This will return
+   * is descending order i.e. will return the most recent runs first. pageToken is used to navigate
+   * to the corresponding "page" of results using <a
+   * href="https://bun.uptrace.dev/guide/cursor-pagination.html">cursor based pagination</a>
+   *
+   * @param limit - how many results the caller wants to be returned, maximum value is 100
+   * @param pageToken - token used for cursor based pagination. if not supplied then the first page
+   *     will be returned
+   * @return ResponseEntity containing the current page of results and a page token for the next
+   *     page if a next page exists
+   */
+  @Override
+  public ResponseEntity<ApiGetPipelineRunsResponse> getAllPipelineRuns(
+      Integer limit, String pageToken) {
+    final SamUser userRequest = getAuthenticatedInfo();
+    String userId = userRequest.getSubjectId();
+
+    int maxLimit = Math.min(limit, 100);
+    CursorBasedPageable cursorBasedPageable = new CursorBasedPageable(maxLimit, pageToken, null);
+    PageSpecification<PipelineRun> pageSpecification =
+        new PageSpecification<>("id", cursorBasedPageable);
+    FieldEqualsSpecification<PipelineRun> userIdSpecification =
+        new FieldEqualsSpecification<>("userId", userId);
+    // grab results from current page based
+    PageResponse<List<PipelineRun>> pageResults =
+        pipelineRunsService.findPageResults(
+            pageSpecification, userIdSpecification, cursorBasedPageable);
```

Comment:
curious why you ask the caller to create the specifications and pageable, rather than passing in just the userId and limit and creating these objects inside the method

Reply:
oh good point, this made everythign so much simpler